File: pkg/day03/day03_test.go
package day03

import (
	"fmt"
	"testing"
)

func TestDay03(t *testing.T) {
	tests := []struct {
		inputPath            string
		wantResult           int
		wantWithConditionals int
	}{
		{inputPath: "testdata/example1.txt", wantResult: 161, wantWithConditionals: 161},
		{inputPath: "testdata/example2.txt", wantResult: 161, wantWithConditionals: 48},
		{inputPath: "testdata/input.txt", wantResult: 189600467, wantWithConditionals: 107069718},
	}

	for _, test := range tests {
		input, err := load(test.inputPath)
		if err != nil {
			t.Fatal(err)
		}

		t.Run(fmt.Sprintf("Part 1: %s", test.inputPath), func(t *testing.T) {
			result := input.process()

			if result != test.wantResult {
				t.Errorf("process: got %d, want %d", result, test.wantResult)
			}
		})

		t.Run(fmt.Sprintf("Part 2: %s", test.inputPath), func(t *testing.T) {
			result := input.process(DO, DONT)

			if result != test.wantWithConditionals {
				t.Errorf("process (with conditionals): got %d, want %d", result, test.wantWithConditionals)
			}
		})
	}
}

File: pkg/day03/day03_test.rb
require "minitest/autorun"
require "minitest/focus"
require "minitest/reporters"

Minitest::Reporters.use!

require_relative "day03"

class Day03Test < Minitest::Test
  def test_part1_example1
    input = Day03.new("testdata/example1.txt")
    assert_equal(161, input.part1)
  end

  def test_part1
    input = Day03.new("testdata/input.txt")
    assert_equal(189600467, input.part1)
  end

  def test_part2_example2
    input = Day03.new("testdata/example2.txt")
    assert_equal(48, input.part2)
  end

  def test_part2
    input = Day03.new("testdata/input.txt")
    assert_equal(107069718, input.part2)
  end
end
